lib/puppet/environments.rb

```diff
@@ -388,13 +390,17 @@ def add_entry(name, cache_entry)
     # Clears the cache of the environment with the given name.
     # (The intention is that this could be used from a MANUAL cache eviction command (TBD)
     def clear(name)
-      @cache.delete(name)
+      entry = @cache.delete(name)
+      if entry
+        @expirations.delete(entry.expires)
+        @next_expiration = @expirations.first || END_OF_TIME
+      end
       Puppet::GettextConfig.delete_text_domain(name)
     end
```

Comment:
The single environment clear doesn't call `Puppet.settings.clear_environment_settings(name)` like `clear_all_expired` does